### PR TITLE
🧹 Clean-ups after #201

### PR DESCRIPTION
**Pull request by favonia**

### CODE REVIEW

File: src/cubical/CofThy.ml [hunk 1]
```diff
@@ -136,40 +137,43 @@ type disj_thy = cached_branches
...
+        | Cof.Meet meet_cofs ->
+          loop (vars, eqs) @@ meet_cofs @ cofs
+        | Cof.Join join_cofs ->
+          join_cofs |> List.concat_map @@ fun join_cof ->
+          loop (vars, eqs) @@ join_cof :: cofs
+        | Cof.Eq (r, s) ->
+          loop (vars, eqs #< (r, s)) cofs
+        | Cof.Neg cof ->
+          loop_neg (vars, eqs) ~neg:cof cofs
+
```
Comment (favonia) [inline src/cubical/CofThy.ml:159]: @jonsterling not sure if you want a blank line here...

**Comment by ecavallo** [comment]
Do you plan to impose a restriction on `hcom`? To ensure there are no empty systems, we'd need to allow `hcom A r r' φ` only when `¬ ¬ φ` is true. `hcom`s that don't satisfy this could be elaborated to `ghcom`s.

It's probably fine to just _always_ elaborate a surface `hcom A r r' {φ}` to the corresponding `hcom A r r' {φ ∨ ¬ φ}`; in the case that `¬ ¬ φ` is already true, we just have `φ ∨ ¬ φ` ⇔ `φ`. (That's in contrast to the old `ghcom`, where treating all surface `hcom`s as `ghcom`s would produce a lot of junk.)

**Comment by favonia** [comment]
> Do you plan to impose a restriction on `hcom`? To ensure there are no empty systems, we'd need to allow `hcom A r r' φ` only when `¬ ¬ φ` is true. `hcom`s that don't satisfy this could be elaborated to `ghcom`s.
> 
> It's probably fine to just _always_ elaborate a surface `hcom A r r' {φ}` to the corresponding `hcom A r r' {φ ∨ ¬ φ}`; in the case that `¬ ¬ φ` is already true, we just have `φ ∨ ¬ φ` ⇔ `φ`. (That's in contrast to the old `ghcom`, where treating all surface `hcom`s as `ghcom`s would produce a lot of junk.)

Unfortunately the solver needs to be fixed first. :facepalm:

**Comment by jonsterling** [comment]
@ecavallo My inclination would be to avoid placing this invariant on `hcom` for now --- but to use this cool `ghcom` in the places where our implementation doesn't preserve the non-falseness of cofibrations. Then the user gets exactly what they ask for: they'll get empty systems exactly when they make them.

I would also be open to imposing the invariant globally in the future but frankly it has always made me a little bit "Geometrically" queasy.

**Comment by favonia** [comment]
Close this now as #200 was not done.

**Pull request closed without merging**